projects/schematics/src/migrations/2_0/removed-public-api.ts

```diff
@@ -0,0 +1,46 @@
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { isImported } from '@schematics/angular/utility/ast-utils';
+import { TODO_SPARTACUS } from '../../shared/constants';
+import { commitChanges, getAllTsSourceFiles, getTsSourceFile, insertCommentAboveIdentifier, InsertDirection } from '../../shared/utils/file-utils';
+import { getSourceRoot } from '../../shared/utils/workspace-utils';
+import { REMOVED_PUBLIC_API_DATA } from './removed-public-api-data';
+
+export function migrate(): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info('Checking removed public api...');
+
+    const project = getSourceRoot(tree, {});
+    const sourceFiles = getAllTsSourceFiles(tree, project);
+    for (const originalSource of sourceFiles) {
+      const sourcePath = originalSource.fileName;
+
+      for (const removedNode of REMOVED_PUBLIC_API_DATA) {
+        if (
+          !isImported(originalSource, removedNode.node, removedNode.importPath)
+        ) {
+          continue;
+        }
+        // 'source' has to be reloaded after each committed change
+        const source = getTsSourceFile(tree, sourcePath);
+
+        const changes = insertCommentAboveIdentifier(
+          sourcePath,
+          source,
+          removedNode.node,
+          buildComment(removedNode.comment ?? `'${removedNode.node}' is no longer part of the public API. Please look into migration guide for more information`)
```

Comment:
Are you referring to `docs/migration/2_0.md`?
Maybe we can think about printing the link either to the customer's console, or as part of every comment (which will probably bring a lot of noise). Take a look at this comment for how the link will look like: https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/6588#issuecomment-593862579

Reply:
As discussed:
- links in each comment brings to much noise
- better to add link to docs as final schematics output